plugin/src/main/java/com/denizenscript/denizen/objects/LocationTag.java

```diff
@@ -4079,6 +4079,41 @@ else if (material.hasModernData() && material.getModernData() instanceof org.buk
             return new ElementTag(BukkitScriptEvent.inCheckInternal(attribute.context, "is_in tag", object, attribute.getParam(), "is_in tag", "is_in tag"));
         });
 
+        // <--[tag]
+        // @attribute <LocationTag.last_interacted_slot>
+        // @returns ElementTag(Number)
+        // @description
+        // Returns the last interacted slot of a Chiseled Bookshelf inventory.
+        // -->
+        tagProcessor.registerTag(ElementTag.class, "last_interacted_slot", (attribute, object) -> {
+            BlockState state = object.getBlockStateForTag(attribute);
+            if (!(state instanceof ChiseledBookshelf)) {
+                return null;
+            }
+            return new ElementTag(((ChiseledBookshelf) object.getBlockState()).getLastInteractedSlot() + 1);
+        });
+
+        // <--[tag]
+        // @attribute <LocationTag.slot[<vector>]>
+        // @returns ElementTag(Number)
+        // @description
+        // Returns the appropriate slot based on a vector relative to this block.
```

Comment:
Slot of what? also I'd probably explain what the vector _is_ a bit more (A point along the block's surface relative to the block's position?)